### PR TITLE
[Docker] Add docker setup

### PR DESCRIPTION
**Pull request by rafryb**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,36 @@
+ARG RUBY_IMAGE=ruby:3.2.2
```
Comment (nukah) [inline Dockerfile:1]: Please use the public aws rep

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+COPY . $APP_ROOT
+WORKDIR $APP_ROOT
+ENV LD_PRELOAD=/usr/lib/libjemalloc.so.2
+
+# Install gems in parallel
+RUN bundle check || \
+  MAKEFLAGS="-j$(nproc)" \
+  bundle install --jobs=$(nproc)
+
+ENTRYPOINT ["script/docker/entrypoint.sh"]
```
Comment (nukah) [inline Dockerfile:36]: I feel that this is taken from some other source, right? 

Can we align it with existing Dockerfile?

Comment (rafryb) [reply Dockerfile:36]: this is related to this topic:
https://github.com/docker/awesome-compose/tree/master/official-documentation-samples/rails/

**Comment by nukah** [comment]
@rafryb main branch was renamed to master (otherwise github workflows will require rewriting)

**Review by nukah — APPROVED** [review]

**Pull request merged**